src/server/status/index.js

```diff
@@ -7,21 +7,40 @@ export default function (kbnServer, server, config) {
   kbnServer.status = new ServerStatus(kbnServer.server);
 
   if (server.plugins['even-better']) {
-    kbnServer.mixin(require('./metrics'));
+    kbnServer.mixin(require('./metrics').collectMetrics);
   }
 
   const wrapAuth = wrapAuthConfig(config.get('status.allowAnonymous'));
-
+  const matchSnapshot = /-SNAPSHOT$/;
   server.route(wrapAuth({
     method: 'GET',
     path: '/api/status',
     handler: function (request, reply) {
+      const v6Format = config.get('status.v6Api');
+      if (v6Format) {
+        return reply({
+          name: config.get('server.name'),
+          uuid: config.get('server.uuid'),
+          version: {
+            number: config.get('pkg.version').replace(matchSnapshot, ''),
+            build_hash: config.get('pkg.buildSha'),
+            build_number: config.get('pkg.buildNum'),
+            build_snapshot: matchSnapshot.test(config.get('pkg.version'))
+          },
+          status: kbnServer.status.toJSON(),
+          metrics: kbnServer.v6Metrics
+        });
+      }
+
       return reply({
```

Comment:
This return should be legacy format with without the nested version information. Metrics also needs to be of the legacy format.